contracts/TokenConverter/AbstractTokenConverter.sol

```diff
@@ -589,72 +518,57 @@ abstract contract AbstractTokenConverter is AccessControlledV8, IAbstractTokenCo
     /// @param tokenAddressOut Address of the token to get after conversion
     /// @param to Address of the tokenAddressOut receiver
     /// @return actualAmountIn Actual amount of tokenAddressIn transferred
-    /// @return amountInMantissa Amount of tokenAddressIn supposed to get transferred
     /// @return actualAmountOut Actual amount of tokenAddressOut transferred
-    /// @return amountConvertedMantissa Amount of tokenAddressOut supposed to get transferred
     /// @custom:error AmountInHigherThanMax error is thrown when amount of tokenAddressIn is higher than amountInMaxMantissa
     function _convertForExactTokens(
         uint256 amountInMaxMantissa,
         uint256 amountOutMantissa,
         address tokenAddressIn,
         address tokenAddressOut,
         address to
-    )
-        internal
-        returns (
-            uint256 actualAmountIn,
-            uint256 amountInMantissa,
-            uint256 actualAmountOut,
-            uint256 amountConvertedMantissa
-        )
-    {
-        (amountConvertedMantissa, amountInMantissa) = getUpdatedAmountIn(
-            amountOutMantissa,
-            tokenAddressIn,
-            tokenAddressOut
-        );
+    ) internal returns (uint256 actualAmountIn, uint256 actualAmountOut) {
+        (, uint256 amountInMantissa) = getUpdatedAmountIn(amountOutMantissa, tokenAddressIn, tokenAddressOut);
 
         if (amountInMantissa > amountInMaxMantissa) {
             revert AmountInHigherThanMax(amountInMantissa, amountInMaxMantissa);
         }
-        (actualAmountIn, actualAmountOut) = _actualAmounts(
-            tokenAddressIn,
-            tokenAddressOut,
-            to,
-            amountInMantissa,
-            amountConvertedMantissa
-        );
+
+        actualAmountIn = _actualAmountIn(tokenAddressIn, amountInMantissa);
+
+        (, amountOutMantissa) = getUpdatedAmountOut(actualAmountIn, tokenAddressIn, tokenAddressOut);
```

Comment:
Given the user is expected an exact amount of tokensOut, if `actualAmountIn < amountInMantissa`, I suppose we should revert the TX, with an error explaining that is not possible to satisfy the request.

Alternatively, we could invoke `_actualAmountIn` with `amountInMaxMantissa` and:
* if `actualAmountIn > amountInMantissa`, refund the difference to the sender, and continue with the function
* if `actualAmountIn == amountInMantissa`, continue with the function
* if `actualAmountIn < amountInMantissa`, revert the TX

In any case, I think we wouldn't need the invocation of `getUpdatedAmountOut` here

Reply:
Invoke `_doTransferIn` with `amountInMaxMantissa` could have undesirable behaviors (like the charge of a huge fee). It seems better to follow with the current approach and do our best